src/POData/ObjectModel/CynicDeserialiser.php

```diff
@@ -89,6 +90,38 @@ protected function isEntryOK(ODataEntry $payload)
         return true;
     }
 
+    protected function isEntryProcessed(ODataEntry $payload, $depth = 0)
+    {
```

Comment:
A payload is completely processed iff all its contained ODataEntry objects have been marked as processed by storing a KeyDescriptor in their id fields, instead of a string.